### ISSUE TITLE
Hotspot plugin - app stop working

### ISSUE THREAD

**Issue by webmaster-imaginet** [hypery2k/cordova-hotspot-plugin#27]
Hi,

**I have a huge issue with the plugin.**

In some of the methods, (especially calling the arp file - getAllHotspotDevices) the plugin caused the app to "freeze" or to "stop working".
I check how it works in the ionic sample app that you uploaded & realized that it has the same issues...

![hotspot_ver1_img1](https://cloud.githubusercontent.com/assets/4920443/14586810/3d1b089c-04ad-11e6-8fb5-c9c399ca13b5.png)
![hotspot_ver1_img2](https://cloud.githubusercontent.com/assets/4920443/14586809/3d1a8548-04ad-11e6-80a2-f7e4d2583022.png)
![hotspot_ver1_img3](https://cloud.githubusercontent.com/assets/4920443/14586811/3d1b0c20-04ad-11e6-8335-9e0ea0486861.png)

I filmed a video to shoe the problems:
[https://www.dropbox.com/s/nej41e10tsw1bhi/hotSpot_ver1.mp4?dl=0](url)

**I really don't know why it is like that, but from a "user experience" point of view, it is very bad & unacceptable situation.**

I try some different approaches to solve this. Unfortunately even using angular promises did not help to solve that matter.

I assume that the plugin need to work in the background. Isn't it?

**Can you please help to solve this problems?**

**Comment by hypery2k** [comment]
I've never experienced such problems. The plugin itself is already running in the background.

Currently did not have much time for this plugin to investigate more ..

**Comment by hypery2k** [comment]
You can try to install the latest master. I've updated it to the dev guide on https://cordova.apache.org/docs/en/4.0.0/guide/platforms/android/plugin.html which should prevent hanging the web thread. Please try and respond in this issue.

**Comment by hypery2k** [comment]
please retry with latest master:

```
cordova plugin add https://github.com/hypery2k/cordova-hotspot-plugin.git
```

and report here again

**Comment by hypery2k** [comment]
tested with Android 5.1.1 on a Nexus 4 and nothing blocks for me.
Published a new plugin version 1.1.1
